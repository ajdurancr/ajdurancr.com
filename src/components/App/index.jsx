import App from 'next/app'
import React from 'react'

class NextApp extends App {
  render() {
    const { Component, pageProps } = this.props
    return (
      <Component {...pageProps} />
    )
  }
}

export default NextApp
