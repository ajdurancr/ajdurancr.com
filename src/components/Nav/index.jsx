import React from 'react'
import './Nav.styles.sass'

const links = [
  { href: '/', label: 'Home' },
  { href: '/about/', label: 'About' },
  { href: '/contact/', label: 'Say Hi!' },
  { href: '/portfolio/', label: 'Portfolio' },
  { href: '/resume/', label: 'Resume' },
  { href: '/blog/', label: 'Blog' },
  { href: 'https://github.com/ajdurancr/durancr-io', label: 'GitHub' },
].map((link) => ({ ...link, key: `nav-link-${link.href}-${link.label}` }))

const Nav = () => (
  <nav>
    <ul>
      {links.map(({ key, href, label }) => (
        <li key={key}>
          <a href={href}>{label}</a>
        </li>
      ))}
    </ul>
  </nav>
)

export default Nav
