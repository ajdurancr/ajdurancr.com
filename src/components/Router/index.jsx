import React from 'react'
import { StaticRouter } from 'react-router'
import { BrowserRouter } from 'react-router-dom'

const isServer = typeof window === 'undefined'

const Router = ({ context, location, children }) => {
  const AppRouter = isServer ? StaticRouter : BrowserRouter

  return (
    <AppRouter location={location} context={context}>
      {children}
    </AppRouter>
  )
}

export default Router
