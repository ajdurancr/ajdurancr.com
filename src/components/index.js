
export { default as Component } from './Component'
export { default as Header } from './Header'
export { default as Menu } from './Menu'
export { default as Page } from './Page'
export { default as Section } from './Section'
export { default as TestComponent } from './TestComponent'
