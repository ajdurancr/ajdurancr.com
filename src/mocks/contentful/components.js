const childComponent = {
  sys: {
    id: '4gfjlQfm3x9ShZKO8gdYqx',
    contentType: {
      sys: {
        id: 'component',
      },
    },
  },
  fields: {
    name: 'ChildComponent',
    contentType: 'Test Component',
    componentDescription: 'This is a child component',
  },
}

export const component = {
  sys: {
    id: '6neuJJsr2mbdKli0VjWnCn',
    contentType: {
      sys: {
        id: 'component',
      },
    },
  },
  fields: {
    name: 'TestComponent',
    contentType: 'Test Component',
    componentDescription: 'This is a test component',
    props: [
      {
        sys: {
          id: '74gEMnRKSOxejniOwjkHYN',
          contentType: {
            sys: {
              id: 'propText',
            },
          },
        },
        fields: {
          name: 'title',
          value: 'This is a test title',
        },
      },
      {
        sys: {
          id: '3hwiQ7OcvmctEdQPye6exu',
          contentType: {
            sys: {
              id: 'propComponents',
            },
          },
        },
        fields: {
          name: 'children',
          value: [
            childComponent,
          ],
        },
      },
      {
        sys: {
          id: '1RezpPBXf2buGYIccqQw0i',
          contentType: {
            sys: {
              id: 'propMarkdown',
            },
          },
        },
        fields: {
          name: 'markdownField',
          value: 'Test value',
        },
      },
    ],
  },
}

export default [component]
