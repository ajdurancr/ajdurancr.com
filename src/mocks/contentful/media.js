export default {
  title: 'Media - GitHub Logo',
  description: 'Github Logo',
  file: {
    url: '//images.ctfassets.net/52uxn1pv2fou/5Yt9H0zn6ewSMnuvzh7V6P/3d6a253c1dce0250c70ef3a52c169872/github.jpg',
    details: {
      size: 11536,
      image: {
        width: 400,
        height: 400,
      },
    },
    fileName: 'GitHub_filename',
    contentType: 'image/jpeg',
  },
}
