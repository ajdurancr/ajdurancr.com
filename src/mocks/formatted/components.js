export const componentProps = {
  name: 'TestComponent',
  contentType: 'Test Component',
  componentDescription: 'This is a test component',
  title: 'This is a test title',
  children: [
    {
      sys: {
        id: '4gfjlQfm3x9ShZKO8gdYqx',
        contentType: {
          sys: {
            id: 'component',
          },
        },
      },
      fields: {
        name: 'ChildComponent',
        contentType: 'Test Component',
        componentDescription: 'This is a child component',
      },
    },
  ],
  markdownField: 'Test value',
}
