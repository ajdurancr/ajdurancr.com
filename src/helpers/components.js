import camelCase from 'lodash.camelcase'
import get from 'lodash.get'

import * as components from '../components'

let componentsMap

const buildComponentsMap = () => Object.keys(components || {})
  .reduce((compMap, componentName) => ({
    ...compMap,
    [camelCase(componentName)]: components[componentName],
  }), {})

export const getComponentsMap = () => {
  if (!componentsMap) {
    componentsMap = buildComponentsMap()
  }


  return componentsMap
}

export const mapProps = (cmsProps) => {
  const { props = [], ...initialProps } = get(cmsProps, 'fields', [])

  return props.reduce((componentProps, { fields }) => ({
    ...componentProps,
    [fields.name]: fields.value,
  }), { ...initialProps })
}
