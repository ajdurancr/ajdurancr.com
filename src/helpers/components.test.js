import React from 'react'

import components from '../components'
import { getComponentsMap, mapProps } from './components'
import { component } from '../mocks/contentful/components'
import { componentProps } from '../mocks/formatted/components'

jest.mock('../components', () => ({}))

const TestComponent = () => <div>Test Component</div>
const ChildComponent = () => <div>Child Component</div>

describe('helpers/components', () => {
  describe('getComponentsMap', () => {
    beforeAll(() => {
      components.TestComponent = TestComponent
      components.ChildComponent = ChildComponent
    })
    it('builds a map with all components', () => {
      expect(getComponentsMap()).toMatchSnapshot()
    })
  })
  describe('mapProps', () => {
    it('returns a props object for a given component', () => {
      expect(mapProps(component)).toEqual(componentProps)
    })
  })
})
