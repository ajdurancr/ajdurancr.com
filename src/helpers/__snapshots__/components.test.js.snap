// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers/components getComponentsMap builds a map with all components 1`] = `
Object {
  "default": Object {
    "ChildComponent": [Function],
    "TestComponent": [Function],
  },
}
`;
